[#3648] docs: Axon Framework 5.0.0-M3 - Expand Getting Started with Event Processing and Handling (#3658)
AxonFramework/AxonFramework: docs/af5-getting-started/modules/ROOT/pages/implement-stateful-event-handler.adoc

        }

        @EventSourcingHandler
        State evolve(StudentSubscribedToCourse event) {
            courses.computeIfPresent(event.courseId(), (id, course) -> course.studentSubscribed());
            return new State(courses, notified);
        }

        @EventSourcingHandler
        State evolve(StudentUnsubscribedFromCourse event) {
            courses.computeIfPresent(event.courseId(), (id, course) -> course.studentUnsubscribed());
            return new State(courses, notified);
        }

        @EventSourcingHandler
        State evolve(AllCoursesFullyBookedNotificationSent event) { // <7>
            return new State(courses, true);
        }
    }
    
    // Command and Event handlers will be added in next steps...
}
----

<1> Simplified to use a single faculty ID for this tutorial
<2> Event-sourced entity that maintains state across multiple events
<3> Inner record representing individual course state with capacity and current enrollment
<4> Business logic to determine if a course is fully booked
<5> Default constructor creating empty state
<6> Event sourcing handlers that evolve state based on different event types

> <7> Handler to track when notification has already been sent - thanks to that you can track it in your EventStore instead of external systems

Reviewer: Minor nit for readability:
```suggestion
<7> Handler to track when notification has already been sent - allows you to track it in your EventStore instead of external systems
```
